src/test/java/org/reactome/release/common/dataretrieval/TestFileRetrieval.java

```diff
@@ -0,0 +1,276 @@
+import org.apache.http.StatusLine;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.conn.ConnectTimeoutException;
+import org.apache.http.entity.ByteArrayEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.protocol.HttpContext;
+import org.junit.Before;
+import org.junit.Test;
+
+import org.mockito.Mock;
+import org.mockito.MockedConstruction;
+import org.mockito.MockedConstruction.Context;
+import org.mockito.MockedConstruction.MockInitializer;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+/**
+ * @author sshorser
+ *
+ */
+public class TestFileRetrieval {
+
+	private static final String MESSAGE_CONTENT = "this is a test";
+	
+	@Mock
+	URI mockUri;
+	
+	@Mock
+	CloseableHttpClient mockClient;
+	
+	@Mock
+	CloseableHttpResponse mockResponse;
+
+	@Mock
+	StatusLine mockStatusLine;
+	
+	HttpEntity entity = new ByteArrayEntity(MESSAGE_CONTENT.getBytes());
+	
+	@Before
+	public void setup()
+	{
+		MockitoAnnotations.openMocks(this);
+	}
+	
+	/**
+	 * Test method for {@link org.reactome.addlinks.dataretrieval.FileRetriever#fetchData()}.
+	 * @throws Exception 
+	 */
+	@Test
+	public void testFetchData() throws Exception
+	{
+		try(MockedStatic<HttpClients> mockedStatic = Mockito.mockStatic(HttpClients.class);)
+		{
+			DataRetriever retriever = new FileRetriever();
+			//retrieve google - it should be pretty easy.
+			URI uri = new URI("http://www.google.com");
+			retriever.setDataURL(uri);
+			
+			Mockito.when(mockResponse.getEntity()).thenReturn(entity);
+			Mockito.when(mockResponse.getStatusLine()).thenReturn(mockStatusLine);
+			Mockito.when(mockClient.execute(any(HttpUriRequest.class))).thenReturn(mockResponse);
+			Mockito.when(mockClient.execute(any(HttpUriRequest.class), (HttpContext) any(HttpContext.class))).thenReturn(mockResponse);
+			
+			Mockito.when(HttpClients.createDefault()).thenReturn(mockClient);
+			
+			String dest = "/tmp/testFetchData_"+ String.valueOf((new Random()).nextInt(Integer.MAX_VALUE));
+			retriever.setFetchDestination(dest);
+			Duration age = Duration.of(5,ChronoUnit.SECONDS);
+			retriever.setMaxAge(age);
+			
+			retriever.fetchData();
+			assertTrue(Files.exists(Paths.get(dest)));
+			
+			//Sleep for two seconds, and then re-download because the file is stale
+			Thread.sleep(Duration.of(2,ChronoUnit.SECONDS).toMillis());
+			retriever.fetchData();
+			assertTrue(Files.exists(Paths.get(dest)));
+			//now set a longer maxAge.
+			age = Duration.of(100,ChronoUnit.SECONDS);
+			retriever.setMaxAge(age);
+			// this time, the file will not be stale (because maxAge is larger) so nothing will be downloaded.
+			retriever.fetchData();
+			//check that the file exists.
+			assertTrue(Files.exists(Paths.get(dest)));
```

Comment:
Should the tests delete the files after they are downloaded and the test is complete so there is a clean state if they are re-run?

Reply:
I think I originally didn't delete them so I could manually verify them, outside of the test.

Reply:
Okay, but this might interfere with re-running the tests?  Maybe do a delete if exists before the tests begin then?

Reply:
Interference seems unlikely, since the filenames include random numbers.